### PR TITLE
This fixes origin callback function types

### PR DESCRIPTION
**Pull request by krisread**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

This pull request updates the Typescript type for the origin callback function as per this issue:

https://github.com/fastify/fastify-cors/issues/53

Without this change, typescript users cannot call `callback(null, false)` which is required behavior.  `null` is not an acceptable value if the first parameter is limited to type `Error`.   Presume this was simply overlooked by the original contributor.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -4,9 +4,9 @@ import { Server, IncomingMessage, ServerResponse } from 'http'
 
 import fastify = require('fastify');
 
-type originCallback = (err: Error, allow: boolean) => void;
+type originCallback = (err?: Error, allow?: boolean) => void;
 
-type originFunction = (origin: string, callback: originCallback) => void;
+type originFunction = (origin: string | undefined, callback: originCallback) => void;
```
Comment (Eomm) [inline index.d.ts]: Why this change?

Comment (krisread) [reply index.d.ts]: I am basing this on the cors / express plugin types, if origin cannot be undefined then we can leave it as a string

**Comment by krisread** [comment]
> Good spot! Would you mind adding a unit test for the change? thanks!

Hi @mcollina  I will probably not be adding a unit test for this as I do not have time / interest in this project - just wanted to be a good citizen and report a bug that will affect every single user in TypeScript. :)  Hope I helped. 

Just an additional note as you may know typescript transpiles to JS at runtime so the type definition is not easily checked in a Javascript test; you would need to add test written in typescript and then it would simply use the type definitions in the test code. 

It seems like you DO have a typescript file that kinda covers this case in your test at `https://github.com/fastify/fastify-cors/blob/master/test/types/index.ts` however despite it being a typescript file your tests are not flagging the obvious type problem; I would say you have setup your test compiling/running incorrectly.  Not something I have time to debug for you guys, sorry. 

Hope my PR and my report help!

**Comment by mcollina** [comment]
No worries, and thanks for the contribution anyway
